include/aspect/coordinate_systems.h

```diff
@@ -45,6 +45,38 @@ namespace aspect
         ellipsoidal,
         invalid
       };
+
+      template<int dim>
+      Tensor<1, dim>
+      get_depth_direction(Point<dim> &position, CoordinateSystem coordinate_system)
+      {
+        Tensor<1, dim> depth_direction;
+        switch (coordinate_system)
+          {
+            case CoordinateSystem::depth:
+            {
+              depth_direction[0] = -1.0;
+              break;
+            }
+            case CoordinateSystem::cartesian:
+            {
+              depth_direction[dim-2] = -1.0;
+
+              break;
+            }
+            case CoordinateSystem::spherical:
+            case CoordinateSystem::ellipsoidal:
+            {
+              depth_direction = position / position.norm();
```

Comment:
I think I don't quite get what the *coordinate system* has to do with it? You seem to be thinking of the coordinate system as a proxy for the kind of geometry you're working in, but that's not how `CoordinateSystem` was intended to be used, I believe, but rather as a way to expressing how a tuple of values should be interpreted as coordinates for a point.